src/modules/cloudflare/captcha/captcha.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Body, Controller, Post, UseGuards, UsePipes } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+
+import { ZodValidationPipe } from '@/shared/utils/zod-validation.pipe';
+
+import { CaptchaService } from './captcha.service';
+import { TokenSchema, TokenSchemaDto } from './dtos/token.dto';
+import { ICaptchaProtectedReturn } from './types';
+import { TurnstileGuard } from '../turnstile/turnstile.guard';
+@ApiTags('captcha')
+@Controller({ path: 'captcha', version: '1' })
+export class CaptchaController {
+  constructor(private readonly captchaService: CaptchaService) {}
+  
+  @UseGuards(TurnstileGuard)
+  @Post('protected')
+  @UsePipes(new ZodValidationPipe(TokenSchema))
+  async captchaProtected(@Body() body: TokenSchemaDto): Promise<ICaptchaProtectedReturn> {
+    return { success: true, message: 'Captcha validation successful! You accessed a protected route.', data: body };
+  }
+
+  @Post('validate-captcha')
+  async validateCaptcha(@Body() body: TokenSchemaDto): Promise<Omit<ICaptchaProtectedReturn, 'data'>> {
+    const result = await this.captchaService.captchaVerification(body);
+    return {
+      message: 'Captcha verification completed.',
+      success: result.success,
+    };
+  }
+}
```

Comment:
We dont need this controller we will use the GUARD on protected router in your case on login route.

Reply:
Example of use

On users controller we put the captcha guard and the frontend need to send the captcha token.

Reply:
![image](https://github.com/user-attachments/assets/348ed3ef-e3ba-44c1-8310-0e20bcdb8865)

Reply:
controller removed